alchemist/alchemist-cognitive-agents/src/main/kotlin/it/unibo/alchemist/model/implementations/groups/GenericGroup.kt

```diff
@@ -0,0 +1,25 @@
+package it.unibo.alchemist.model.implementations.groups
+
+import it.unibo.alchemist.model.interfaces.Group
+import it.unibo.alchemist.model.interfaces.Node
+
+/**
+ * Basic implementation of a group.
+ */
+open class GenericGroup<T, N : Node<T>> : Group<T, N> {
+
+    private val _members: MutableList<N> = mutableListOf()
```

Comment:
Prefer immutable var over mutable val + backing property

Reply:
I couldn't do anything else in that case because groups are populated on pedestrians creation so I need to modify them on the fly.

Reply:
Yes you can :)
you can replace the `val` of type `mutableList` with a `var` of type `List` (read-only). You can replace the entire list every time.

However in this very case, even though it would be cleaner, performances would be worse. Let's keep it as it is.